### PR TITLE
Update documentation, esp. guidance on how to use pykale

### PR DESCRIPTION
**Pull request by haipinglu**
### Description

Update documentations to provide a clearer guidance on how to use pykale, including a very brief "tutorial", which is not in the ideal form but should be better than nothing. Hope this could help potential users interested in using pykale. This is motivated by a [feedback from _LargeYellowBus_ on reddit](https://www.reddit.com/r/MachineLearning/comments/mxtra6/p_an_introduction_to_pykale/) and incorporated contents from the recent slides.

_Note_: For reviews, please focus on the contents, hyperlinks may only be active after merging.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Review by pz-white — APPROVED** [review]
The new guidance LGTM!

**Review by XianyuanLiu — COMMENTED** [review]
Looks great. Thanks!

File: docs/source/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Usage of Pipeline-based API in Examples
+
+The `kale` API has a unique pipeline-based API design. Here we use the [domain adaptation for digits classification example](https://github.com/pykale/pykale/tree/main/examples/digits_dann_lightn) to explain our design. Each example contains through essential modules and one directory:
+
+- `main.py` is the main module to be run, showing the main workflow.
+- `config.py` is the configuration module setting the data, prediction problem, and hyper-parameters, etc. The settings in this module is the default configuration.
+  - `configs` is the directory to place the *customized* configurations for each individual run. We use `.yaml` files for this purpose.
+- `model.py` is the  model module to define the machine learning model and configure its training parameters.
+- `trainer.py` is the trainer module to define the trainer, *only needed when not using `pytorch lightning`*.
```
[the remark this answers is not in the document]
Comment (pz-white) [reply docs/source/tutorial.md:11]: I think it should be fine as the trainer module is only required without using `pl`?

File: README.md [hunk 2]
```diff
@@ -27,43 +27,53 @@
...
 ```bash
 pip install pykale
 ```
 
 For more details and other options, please refer to [the installation guide](https://pykale.readthedocs.io/en/latest/installation.html).
 
-## Examples, Tutorials, and Discussions
+### Step 1: Tutorial and Examples
+
+Start with a brief [tutorial](https://pykale.readthedocs.io/en/latest/tutorial.html#usage-of-pipeline-based-api-in-examples) walking through API usage in examples.
```
[the remark this answers is not in the document]
Comment (pz-white) [reply README.md:50]: I also meet the same issue.

[the remark this answers is not in the document]
Comment (pz-white) [reply README.md:50]: It should be accessed after this PR merging and re-build doc? As I see the tutorial is added below in the doc's .rst, so we cannot access it now.

**Review by pz-white — APPROVED** [review]

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (haipinglu) [reply README.md:50]: > Something wrong with this link. Is it my own problem?

Yes, link will only be active after merging. You can find the content in markdown.

File: docs/source/tutorial.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (haipinglu) [reply docs/source/tutorial.md:11]: > There is no `trainer.py` in `digits_dann_lightn`. Do we need to add one?

I said *only needed when not using `pytorch lightning`*. `digits_dann_lightn` is using lightning.

**Review by sz144 — APPROVED** [review]

**Pull request merged**